### PR TITLE
[13.0][MIG] project_timeline: Migration to 13.0

### PR DESCRIPTION
**Pull request by Tardo**
**Dependens on** https://github.com/OCA/web/pull/1557

cc @tecnativa TT19798

### CODE REVIEW

File: project_timeline/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+* Pedro M. Baeza <pedro.baeza@tecnativa.com>
+* Carlos Dauden <carlos.dauden@tecnativa.com>
+* Alexandre Moreau <alexandre.moreau@doscaal.fr>
+* Dennis Sluijk <d.sluijk@onestein.nl>
+* Nikul Chaudhary <nikulchaudhary2112@gmail.com>
+* Eduardo Magdalena <emagdalena@c2i.es> (C2i Change 2 improve http://www.c2i.es)
+* Alexandre Díaz <alexandre.diaz@tecnativa.com>
```
Comment (manuelcalerosolis) [inline project_timeline/readme/CONTRIBUTORS.rst]: * `Tecnativa <https://www.tecnativa.com>`__:

  * Pedro M. Baeza
  * Carlos Dauden 
  * Alexandre Díaz

**Review by manuelcalerosolis — APPROVED** [review]

File: project_timeline/readme/CONTRIBUTORS.rst (hunk 1 above)
Comment (Tardo) [reply project_timeline/readme/CONTRIBUTORS.rst]: Ummm is not a problem breaking the contribution history?
Now i know that Pedro was the first person in contributing...

Comment (pedrobaeza) [reply project_timeline/readme/CONTRIBUTORS.rst]: Well, this list doesn't intend to be contribution in historical sort. What happens if I contribute the initial version and then again 5 times later? The copyright in files tend to be more accurate, expressing years. Any way, the only trustable source is the commit history. That's why we assure always that is preserved.

**Comment by Tardo** [comment]
Changes done!

**Review by dreispt — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@Tardo Travis is red. Can you check (maybe oca_dependencies.txt)?

**Comment by Saran440** [comment]
Functional Test
- demo error when install project_timeline module
Selection_011

- click icon timeline but it not works
Peek 2020-05-06 10-03

**Comment by pedrobaeza** [comment]
Have you tried locally or in runbot? I suspect that the PR is not rebased, and thus it won't work in runbot

**Comment by Saran440** [comment]
@pedrobaeza I tested it on localhost, new database with demo and
my code from git checkout branch Tecnativa:13.0-mig-project_timeline

**Comment by pedrobaeza** [comment]
OK, let's see what @Tardo says about it

**Comment by tonygalmiche** [comment]
> @pedrobaeza I tested it on localhost, new database with demo and
> my code from git checkout branch Tecnativa:13.0-mig-project_timeline

I just tested on my computer and I have the same problems

**Comment by mikolodz** [comment]
I have also tried on my local instance, with the same result - view icon is not actually changing the project view. Installed modules without any problems.


odoo-13-project-timeline-view-not-showing-up

**Comment by Tardo** [comment]
Reported issues fixed.
Related fixes PR: https://github.com/OCA/web/pull/1598

**Comment by mikolodz** [comment]
> Reported issues fixed.
> Related fixes PR: [OCA/web#1598](https://github.com/OCA/web/pull/1598)

After this commit, for me the view is loading, but while opening and then while trying to move the view (click, hold, move), I'm getting this error:

```
TypeError: undefined is not an object (evaluating 'datas._data[key]')

https://xxx/web/content/1556-70a2fde/web.assets_backend.js:3933
Traceback:
draw_dependencies@https://xxx/web/content/1556-70a2fde/web.assets_backend.js:3933:98
draw_canvas@https://xxx/web/content/1556-70a2fde/web.assets_backend.js:3932:527
https://xxx/web/content/1556-70a2fde/web.assets_backend.js:3932:428
https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:17815:28
[native code]
_redraw@https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:24837:30
_redraw@https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:37574:35
_redraw@[native code]
https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:23787:26
https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:17815:28
[native code]
_redraw@https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:24827:34
_redraw@https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:37574:35
_redraw@[native code]
https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:23787:26
https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:17815:28
[native code]
_redraw@https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:24827:34
_redraw@https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:37574:35
_redraw@[native code]
https://xxx/web_timeline/static/lib/vis-timeline/vis-timeline-graph2d.js:9006:12
```

odoo-13-web-timeline-error

**Comment by Tardo** [comment]
@mikolodz Hi, i can't reproduce this latest issue:
rec

**Comment by mikolodz** [comment]
> @mikolodz Hi, i can't reproduce this latest issue:

Ok, thanks, I will try to reproduce this issue using the latest docker image build, it may be connected with something else (although I didn't hit any bug in other modules yet).

**Review by tonygalmiche — CHANGES_REQUESTED** [review]
I just tested this last commit with latest docker image
I have no error during installation
I don't have a javascript error when clicking on the view
But the timeline view is not displayed
So it doesn't work

**Comment by mikolodz** [comment]
> I just tested this last commit with latest docker image
> I have no error during installation
> I don't have a javascript error when clicking on the view
> But the timeline view is not displayed
> So it doesn't work

After I uninstalled module **Project Timeline Task Dependencies**, I get no longer the error while clicking&dragging.
In the Project timeline view (not task) I get an empty screen.

BTW. I updated Odoo docker image to the lastest - no change in the behaviour.
odoo-13-project-timeline-error

**Comment by tonygalmiche** [comment]
with this https://github.com/OCA/web/pull/1599, it works, but I have this icon : 
image

**Comment by chrisandrewmann** [comment]
> 
> 
> with this [OCA/web#1599](https://github.com/OCA/web/pull/1599), it works, but I have this icon :
> image

Hi @tonygalmiche, I didn't realise that @Tardo had already done a PR to fix this but mine worked ok to fix it for me, he's welcome to decide which one to take. I also deliberately changed the icon to the hourglass as I felt that works best, it's not used anywhere else in Odoo and the Tasks icon used before is confusing in Enterprise as it's identical to the Gantt view. Feel free to suggest a different icon if you like but I think it should be unique across all Odoo view buttons.

**Comment by tonygalmiche** [comment]
Hi @chrisandrewmann,
Neither did I see that it was necessary to use this PR :)
https://github.com/OCA/web/pull/1557
so I just tried it again with this PR, but it doesn't work
but it works with yours : 
https://github.com/OCA/web/pull/1599

personally I prefer this icon : 
image

File: project_timeline/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+* Go to *Project > Search > Tasks* or *Project > Dashboard*.
+* Click on the icon with a clock.
```
Comment (Saran440) [inline project_timeline/readme/USAGE.rst]:
```suggestion
* Click on the icon with a timeline.
```
please change it because icon clock is view activity.

**Review by Saran440 — COMMENTED** [review]
Functional Test, LGTM :+1: But I don't know is it bug?
- In v.12, I can mouse scroll up/down for zome in/out. But v.13 can't

**Comment by pedrobaeza** [comment]
You must use OCA/web#1598 that was published first and including other improvements. Please confirm it's working with it. I also agree the icon is correct. This is a replacement for enterprise gantt, so it's OK to use the same icon.

**Review by tonygalmiche — APPROVED** [review]
Thank you @pedrobaeza,
I did not understand that it was necessary to use this PR https://github.com/OCA/web/pull/1598
I just tested again with this PR and everything works fine and I have the right icon :)
Sorry for my mistake

**Comment by pedrobaeza** [comment]
Thanks to you for checking :)

File: project_timeline/readme/USAGE.rst (hunk 2 above)
Comment (tonygalmiche) [reply project_timeline/readme/USAGE.rst]: > Functional Test, LGTM +1 But I don't know is it bug?
> 
>     * In v.12, I can mouse scroll up/down for zome in/out. But v.13 can't

I confirm that the zoom does not work even with the PR https://github.com/OCA/web/pull/1598

But I don't know if it's a bug either :)

Comment (pedrobaeza) [reply project_timeline/readme/USAGE.rst]: That's a vislibrary feature that may have changed, but has nothing to do with the modules themselves.

**Review by Saran440 — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Incorrect close...

**Pull request merged**